pkg/ipblock/ipblock.go

```diff
@@ -236,19 +242,15 @@ func FromCidrList(cidrsList []string) (*IPBlock, error) {
 	return res, nil
 }
 
-// FromCidrExcept returns an IPBlock object from input cidr str an exceptions cidr str
-func FromCidrExcept(cidr string, exceptions []string) (*IPBlock, error) {
-	span, err := cidrToInterval(cidr)
-	if err != nil {
-		return nil, err
-	}
-	ipRange := interval.FromInterval(span.Start, span.End)
+// Except returns an IPBlock object from input cidr str an exceptions cidr str
+func (b *IPBlock) Except(exceptions ...string) (*IPBlock, error) {
+	ipRange := b.ipRange.Copy()
 	for i := range exceptions {
 		intervalHole, err := cidrToInterval(exceptions[i])
 		if err != nil {
 			return nil, err
 		}
-		ipRange.AddHole(intervalHole)
+		ipRange = ipRange.Subtract(interval.FromInterval(intervalHole.Start, intervalHole.End))
```

Comment:
Code might be clearer if you create IPBlocks from the cidrs and call `IPBlock.Subtract`

Reply:
Also more efficient, since the copy only happens once. Done.